test/units/playbook/test_collectionsearch.py

```diff
@@ -0,0 +1,38 @@
+# (c) 2020 Ansible Project
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.playbook.collectionsearch import CollectionSearch
+
+import pytest
+
+
+def test_collection_static_warning(capsys):
+    """Test that collection name is not templated.
+
+    Also, make sure that users see the warning message for the referenced name.
+    """
+
+    collection_name = 'foo.{{bar}}'
+    cs = CollectionSearch()
+    assert collection_name in cs._load_collections(None, [collection_name])
+
+    std_out, std_err = capsys.readouterr()
+    assert '[WARNING]: "collections" is not templatable, but we found: %s' % collection_name in std_err
```

Comment:
I'd use equality which would be more precise and replace that check that you had earlier with calls count == 1.

Reply:
I was using IN here because it seems the capsys output contains color-coding chars and newlines so IN was easier to use for confirming the main data here (warning + collection name) is present.

Reply:
Right, you could probably monkeypatch `ANSIBLE_COLOR`: https://github.com/ansible/ansible/blob/598ead965dd0ed1db9a554e8a02c8fcc693ffdd8/lib/ansible/utils/color.py#L91.

But I looked closer at the display function and it looks like it uses an actual logger from `logging` stdlib module (https://github.com/ansible/ansible/blob/598ead965dd0ed1db9a554e8a02c8fcc693ffdd8/lib/ansible/utils/display.py#L216). So it's even easier than I thought initially.

You should be able to use the [`caplog` fixture](https://docs.pytest.org/en/latest/logging.html#caplog-fixture). Let me see if I can quickly come up with a patch.